feat: add internationalization

maltejur opened the pull request:
TODO before merge:

- [x] update all strings to use translations
- [x] add language selector
- [x] (use explanations from other languages)

maltejur committed:
Add next-i18next

maltejur committed:
Add spanish

weblate committed:
Translations update from Weblate (#28)

* Added translation using Weblate (Spanish)

* Translated using Weblate (Spanish)

Currently translated at 25.0% (1 of 4 strings)

Translation: The Algorithms Website/Index
Translate-URL: https://hosted.weblate.org/projects/TheAlgorithms/index/es/

Co-authored-by: Malte Jürgens <maltejur@web.de>

maltejur committed:
Fix build error

weblate committed:
feat: Translations update from Weblate (#29)

* Added translation using Weblate (Spanish)

* Translated using Weblate (Spanish)

Currently translated at 25.0% (1 of 4 strings)

Translation: The Algorithms Website/Index
Translate-URL: https://hosted.weblate.org/projects/TheAlgorithms/index/es/

* Added translation using Weblate (Norwegian Bokmål)

* Added translation using Weblate (Turkish)

Co-authored-by: Malte Jürgens <maltejur@web.de>
Co-authored-by: Allan Nordhøy <epost@anotheragency.no>
Co-authored-by: Oğuz Ersen <oguzersen@protonmail.com>

Panquesito7 committed:
feat: Add support for more translations and strings (#30)

Panquesito7 committed:
Merge branch 'main' into add_internationalization

maltejur committed:
fix: fix eslint errors

maltejur committed:
Merge branch 'add_internationalization' of https://github.com/TheAlgorithms/website into add_internationalization

Panquesito7 commented on public/locales/de/index.json:
```diff
@@ -0,0 +1,6 @@
+  "algorithmExplanation": "Ein Algorithmus ist ein Satz von Regeln, der eine oder mehrere Eingaben entgegennimmt, dann innere Berechnungen und Datenmanipulationen durchführt und eine Ausgabe oder einen Satz von Ausgaben zurückgibt. Kurz gesagt: Algorithmen machen das Leben einfach. Von komplexen Datenmanipulationen und Hashes bis hin zu einfacher Arithmetik folgen Algorithmen einer Reihe von Schr ...
+  "aboutUsTitle": "Über uns",
+  "aboutUs": "Wir sind eine Gruppe von Programmierern, die sich gegenseitig helfen, neue Dinge zu bauen, sei es das Schreiben komplexer Verschlüsselungsprogramme oder einfacher Chiffren. Unser Ziel ist es, zusammenzuarbeiten, um schöne, hilfreiche und interessante Algorithmen mit Code zu dokumentieren und zu modellieren. Wir sind eine Open-Source-Gemeinschaft - jeder kann etwas beitragen. Wir übe ...
+}
```
```suggestion
}

```

maltejur committed:
Rename "index.json" files to "common.json"

Panquesito7 committed:
feat: Translate the categories list

An Spanish string was added to test if it works.

Panquesito7 committed:
feat: Translate the `About` text in the `navbar`

An Spanish string was added to test if it works.

Panquesito7 committed:
fix: Errors reported by Codacy

Panquesito7 committed:
feat: Add translation support for more strings

Some Spanish strings were added to test if it works properly.

maltejur committed:
Add donateText

Panquesito7 committed:
feat: Add translation for the donation text

maltejur committed:
fix error with 404 errors on other locales

maltejur committed:
Merge branch 'add_internationalization' of https://github.com/TheAlgorithms/website into add_internationalization

maltejur committed:
Update README.md

Panquesito7 commented on README.md:
```diff
@@ -25,10 +25,22 @@
+
+[![Global translation status](https://hosted.weblate.org/widgets/TheAlgorithms/-/index/287x66-white.png)](https://hosted.weblate.org/engage/TheAlgorithms/?utm_source=widget)
+
+[![Translation status by language](https://hosted.weblate.org/widgets/TheAlgorithms/-/index/multi-auto.svg)](https://hosted.weblate.org/projects/TheAlgorithms/common/?utm_source=widget)
```
```suggestion
[![Translation status by language](https://hosted.weblate.org/widgets/TheAlgorithms/-/common/multi-auto.svg)](https://hosted.weblate.org/projects/TheAlgorithms/common/?utm_source=widget)
```

Panquesito7 committed:
fix: Apply suggestions from code review

maltejur committed:
Translated using Weblate (German)

Currently translated at 66.6% (20 of 30 strings)

Translation: The Algorithms/Common
Translate-URL: https://hosted.weblate.org/projects/TheAlgorithms/common/de/

Panquesito7 committed:
Translated using Weblate (Spanish)

Currently translated at 90.0% (27 of 30 strings)

Translation: The Algorithms/Common
Translate-URL: https://hosted.weblate.org/projects/TheAlgorithms/common/es/

weblate committed:
Merge remote-tracking branch 'origin/add_internationalization' into add_internationalization

weblate committed:
fix: Translations update from Weblate (#32)

* Translated using Weblate (German)

Currently translated at 66.6% (20 of 30 strings)

Translation: The Algorithms/Common
Translate-URL: https://hosted.weblate.org/projects/TheAlgorithms/common/de/

* Translated using Weblate (Spanish)

Currently translated at 90.0% (27 of 30 strings)

Translation: The Algorithms/Common
Translate-URL: https://hosted.weblate.org/projects/TheAlgorithms/common/es/

Co-authored-by: Malte Jürgens <maltejur@web.de>
Co-authored-by: David Leal <halfpacho@gmail.com>

fitojb committed:
Translated using Weblate (Spanish)

Currently translated at 90.0% (27 of 30 strings)

Translation: The Algorithms/Common
Translate-URL: https://hosted.weblate.org/projects/TheAlgorithms/common/es/

maltejur committed:
Merge branch 'add_internationalization' of https://github.com/TheAlgorithms/website into add_internationalization

maltejur committed:
fix wrong manual merge

Panquesito7 commented:
> * [ ] (use explanations from other languages)

The Spanish translation for **all** English-based algorithms is done. https://github.com/TheAlgorithms/Algorithms-Explanation/pull/93

maltejur committed:
First implementation of explanation localization

Panquesito7 committed:
feat: Add Exponential Search in `lib/aliases.js`

Panquesito7 commented on yarn.lock:
```diff
@@ -640,10 +682,10 @@ ansi-styles@^4.0.0, ansi-styles@^4.1.0:
-  version "5.0.0"
-  resolved "https://registry.yarnpkg.com/ansi_up/-/ansi_up-5.0.0.tgz#9bc3b087a59bd543c98dd235a9c8e0a842c454e9"
-  integrity sha512-RHw/w3Kb2U3k4XKfl8FXZW9ldxtTBbLNdKO0RboYeU4ReVwRP77M7b/OxiavMGZsBWcDxn/T0QiR+VtLf7mPYw==
+ansi_up@^4.0.4:
```
`ansi_up` should be at least version 5.0.0, as in the `4.0.4` version there's a medium-severity vulnerability.

maltejur replied:
> \>yarn audit
yarn audit v1.22.10
0 vulnerabilities found - Packages audited: 786

Yarn tells me everything is fine. Where did you find this?

Panquesito7 replied:
Dependabot reported an error previously. Let's see if that happens when we merge this PR...

maltejur replied:
okay

maltejur committed:
Fix eslint errors

maltejur committed:
Add language select

maltejur committed:
Add translation icon on mobile

weblate committed:
Translations update from Weblate (#33)

* Translated using Weblate (German)

Currently translated at 93.9% (31 of 33 strings)

Translation: The Algorithms/Common
Translate-URL: https://hosted.weblate.org/projects/TheAlgorithms/common/de/

* Translated using Weblate (German)

Currently translated at 100.0% (33 of 33 strings)

Translation: The Algorithms/Common
Translate-URL: https://hosted.weblate.org/projects/TheAlgorithms/common/de/

Co-authored-by: Malte Jürgens <maltejur@web.de>

weblate committed:
Translations update from Weblate (#34)

* Translated using Weblate (German)

Currently translated at 93.9% (31 of 33 strings)

Translation: The Algorithms/Common
Translate-URL: https://hosted.weblate.org/projects/TheAlgorithms/common/de/

* Translated using Weblate (German)

Currently translated at 100.0% (33 of 33 strings)

Translation: The Algorithms/Common
Translate-URL: https://hosted.weblate.org/projects/TheAlgorithms/common/de/

Co-authored-by: Malte Jürgens <maltejur@web.de>

maltejur committed:
Update country flags

maltejur committed:
Merge branch 'add_internationalization' of https://github.com/TheAlgorithms/website into add_internationalization

maltejur committed:
Fix error with language select

weblate committed:
Translated using Weblate (Spanish) (#36)

Currently translated at 90.9% (30 of 33 strings)

Translation: The Algorithms/Common
Translate-URL: https://hosted.weblate.org/projects/TheAlgorithms/common/es/

Co-authored-by: David Leal <halfpacho@gmail.com>

Panquesito7 reviewed (approved):
Amazing work; thank you! This works perfectly. Can be merged. 👍 🎉
